shared/templates/kernel_build_config/oval.template

```diff
@@ -0,0 +1,30 @@
+<def-group>
+  <definition class="compliance" id="{{{ _RULE_ID }}}"
+  version="1">
+    {{{ oval_metadata("The kernel " + CONFIG + " should have value" + VALUE, affected_platforms=["multi_platform_all"]) }}}
+    <criteria>
+      <criterion comment="Check build configuration of installed kernels"
+      test_ref="test_kernel_{{{ CONFIG}}}" />
+    </criteria>
+  </definition>
+
+  <ind:textfilecontent54_test check="all" check_existence="{{{ EXISTENCE }}}" comment="Check /boot/config-.* files for {{{ CONFIG }}}={{{ VALUE }}}"
+  id="test_kernel_{{{ CONFIG}}}" version="1">
+    <ind:object object_ref="object_kernel_{{{ CONFIG }}}" />
+{{% if EXISTENCE != "none_exist" %}}
+    <ind:state state_ref="state_kernel_{{{ CONFIG }}}" />
+{{% endif %}}
+  </ind:textfilecontent54_test>
+
+  <ind:textfilecontent54_object id="object_kernel_{{{ CONFIG }}}" version="1">
+    <ind:filepath operation="pattern match">^/boot/config-.*$</ind:filepath>
+    <ind:pattern operation="pattern match">^{{{ CONFIG | upper }}}=(.*)$</ind:pattern>
```

Comment:
The `upper` function should be at least documented. Or maybe, remove it and use directly the input from rule.yml files (with correct upper case already, it should be more transparent).

Reply:
If using the upper case, you can do the reverse by using `lower` function in the identifiers or have the template.py create a new attribute coming from `config` with `lower`

Reply:
I lean towards having the input in the rule lower case format and make all capital letters when needed. It is annoying to write with all caps.

Reply:
Hmm, anyway, input in upper case seems more natural for these configs.